diagnostics/example_multicase/example_multicase.py

```diff
@@ -0,0 +1,155 @@
+# MDTF Example Diagnostic POD for Multiple Cases / Experiments
+# ================================================================================
+#
+# This file is part of the Multicase Example Diagnostic POD of the MDTF code
+# package (see mdtf/MDTF-diagnostics/LICENSE.txt)
+#
+# Example Diagnostic POD
+#
+#   Last update: Feb-2022
+#
+#   This example builds upon the single case (i.e. `example`) example POD
+#   and illustrates how to design and implement a POD that uses multiple
+#   model source datasets. These can be the same experiment with different
+#   models, two different experiments from the same model, or two different
+#   time periods within the same simulation.
+#
+#   Version & Contact info
+#
+#   - Version/revision information: version 1 (Feb-2022)
+#   - Model Development Task Force Framework Team
+#
+#   Open source copyright agreement
+#
+#   The MDTF framework is distributed under the LGPLv3 license (see LICENSE.txt).
+#
+#   Functionality
+#
+#   Metadata associated with all of the different cases are passed from the
+#   framework to the POD in the form of a dictionary. The POD iterates over the
+#   case list and opens the input datasets.  The `tas` variable is extracted for
+#   each case and the time average is taken over the dataset. Anomalies are calculated
+#   relative to the global mean and then zonally-averaged. The resulting plot
+#   contains one line for each case.
+#
+#   Required programming language and libraries
+#
+#     * Python >= 3.7
+#     * xarray
+#     * matplotlib
+#
+#   Required model output variables
+#
+#     * tas - Surface (2-m) air temperature (CF: air_temperature)
+#
+#   References
+#
+#      Maloney, E. D, and Co-authors, 2019: Process-oriented evaluation of climate
+#         and wether forcasting models. BAMS, 100(9), 1665-1686,
+#         doi:10.1175/BAMS-D-18-0042.1.
+
+
+# Import modules used in the POD
+import os
+import matplotlib
+
+matplotlib.use("Agg")  # non-X windows backend
+
+import numpy as np
+import xarray as xr
+import matplotlib.pyplot as plt
+
+# Part 1: Read in the model data
+# ------------------------------
+
+# Recieve a dictionary of case information from the framework. For now, we will
+# "fake" a dictionary now with information we are getting from the single case
+# POD that is processed by the framework
+
+case_info = {
```

Comment:
This is a good reference, since it make sense to make the case list available to the POD to parse from jsonc file (could be a separate flle, or part of POD's settings.jsonc file). The framework [launches PODs as a subprocess](https://github.com/NOAA-GFDL/MDTF-diagnostics/blob/858303d4d33455c154846ec5cf8749564afb06e1/src/environment_manager.py#L443-L469), and specifies the environment variables using the `env=env_vars` option. Subprocess does allow passing arguments to programs that it calls, but it does not seem like it can handle the nested dictionary structure we're working.

The only way to pass the case_list dictionary from the framework to the POD would be with [multiprocessing](https://developpaper.com/data-sharing-in-multithreading-and-multiprocessing-in-python/). This method is more complex, and carries the risk of overloading system resources and race conditions. 

Whether the case_list comes from the framework or directly from a jsonc file (or yaml), the POD would need a method to receive case_list. Users could obtain the case_list from a file using the [read_json](https://github.com/wrongkindofdoctor/MDTF-diagnostics/blob/38352248bccf25e91590fd3b339b29b138c5c26c/src/util/filesystem.py#L341-L366) routine in src/util/filesystem.py:
```
from src.util import read_json
    code_root = "/home/mdtf/MDTF-diagnostics"
    file_name = "src/default_tests.jsonc"
    json_obj = read_json(os.path.join(code_root, file_name)'
    case_list = json_obj.get('case_list')

```
Using multiprocessing:
```
# Framework routine following the example in the multprocessing link
#  (not entirely sure if this is correct)
import multiprocessing
p = multiprocessing.Process(target=example.py, args=(case_list))
p.start()
p.join()
````

```
# POD case_list intake routine
def read_case_list(**case_list):
   # access each dictionary element
for k, v in case_list.items():
   # display case name
    print(k)
    for k2 in v:
            # case name attribute keys and values
            print(k2,v[k2])
```

Reply:
@jkrasting @wrongkindofdoctor great ideas and a nice motivating POD example for the multi-run capability. I started a new [discussion](https://github.com/NOAA-GFDL/MDTF-diagnostics/discussions/315) on this topic, please chime in, and we could bring it up at the mini-leads meeting as well.